docs/universal-gateway/examples/maintenance-mode.mdx

```diff
@@ -0,0 +1,76 @@
+---
+title: "Deploy a Zero-Code Maintenance Mode for Your Services"
+sidebar_label: Maintenence Mode
+description: Learn how to use Traffic Policy to immediately deploy a consistent maintenance message without needing to spin up a new backend service or change DNS records.
+---
+
+import ReserveDomain from "./snippets/_reserve-domain.mdx";
+import CloudEndpoint from "./snippets/_cloud-endpoint.mdx";
+import TryOut from "./snippets/_try-out.mdx";
+import Back from "./snippets/_back-to-examples.mdx";
+
+Whenever you're making infrastructure changes, a major upgrade to your services during planned downtime, or experiencing an incident, you still want to maintain a public-facing presence on your endpoints to tell customers what's going on.
+
+With ngrok, you can use Traffic Policy and the [`custom-response`](https://ngrok.com/docs/traffic-policy/actions/custom-response/) action to immediately deploy a consistent maintenance message without needing to spin up a new backend service or change DNS records to point to an third-party app.
+
+## 1. Create an endpoint for your service
+
+Start an internal [Agent Endpoint](/docs/universal-gateway/agent-endpoints/), replacing `$PORT` based on where your service listens.
+You can also use one of our [SDKs](/docs/agent-sdks) or the [Kubernetes Operator](/docs/k8s).
+
+```bash
+ngrok http $PORT --url https://service.internal
+```
+
+## 2. Reserve a domain
+
+<ReserveDomain />
+
+## 3. Create a Cloud Endpoint
+
+<CloudEndpoint />
+
+## 4. Create a maintenance page on your Cloud Endpoint with Traffic Policy
+
+While still viewing your new cloud endpoint in the dashboard, copy and paste the policy below into the Traffic Policy editor.
+
+```yaml
+on_http_request:
+  - actions:
+      - type: custom-response
+        config:
+          body: "We're currently undergoing maintenance. Please check back later!"
+  - actions:
+      - type: forward-internal
+        config:
+          url: https://service.internal
+```
+
+### What's happening here?
+
+This policy "interrupts" every request by providing a custom response containing your maintenance message.
+
+When your maintenance window is over and you're ready to route traffic to your upstream service, remove or comment out the `custom-response` action.
+
+```yaml
+on_http_request:
+  #- actions:
+  #    - type: custom-response
+  #      config:
+  #        body: "We're currently undergoing maintenance. Please check back later!"
+  - actions:
+      - type: forward-internal
+        config:
+          url: https://service.internal
+```
+
+## 5. Try out your endpoint
+
+<TryOut />
+
+## What's next?
+
+- Read more about [Traffic Policy](/docs/traffic-policy) and its [core concepts](/docs/traffic-policy/concepts).
```

Comment:
```suggestion
- Learn more about using Traffic Policy to return custom responses by [trying out our examples](/docs/traffic-policy/actions/custom-response/#examples).
```
I think it might be more useful for us to point them directly to the parts of the Traffic Policy docs that are pertinent to what's being discussed on this page.